test/test_manager.cpp

```diff
@@ -39,19 +40,19 @@ void testComponentShouldBeRestartedAfterRunError() {
 
     // We need to be able to make sure a restart was done
     testComponent.set_initialize_return({.ok      = true,
-                                         .message = "",
+                                         .error = {},
                                          .called  = false});
 
     testComponent.set_run_return({.ok      = false,
-                                  .message = "forced error",
+                                  .error = std::make_error_code(ESP_FAIL),
                                   .called  = false});
 
     sleep(1);
 
     while (!testComponent.run_called()) {}
 
     testComponent.set_run_return({.ok      = true,
-                                  .message = "",
+                                  .error = std::make_error_code(ESP_FAIL),
```

Comment:
```suggestion
                                  .error = {},
```
You did run the unit tests right? :^)